### ISSUE TITLE
Restore select to initial value

### ISSUE THREAD

**Issue by pieecia** [Choices-js/Choices#1023]
hey,

I checked the live preview in the documentation, the button to reset the form (`button type="reset"`) causes select not to keep the default value, select is empty

this feature works fine in version 9.0.1, the bug has appeared since version 9.1.0

how to reproduce the issue:

- visit https://choices-js.github.io/Choices/
- change select to a random value
- click the "reset" button

![Zrzut ekranu 2022-03-21 o 11 26 45](https://user-images.githubusercontent.com/8327453/159243393-79ad3b0f-6a30-4303-89e9-2c9b7375b0fa.png)

**Comment by marcelaerts** [comment]
Is there a solution for this issue?

**Comment by yanmorinokamca** [comment]
We experienced the same problem. We call form.reset() between each page load. 
Seems like form.reset() call Choices._onFormReset -> resetTo with _initialState.
_initialState doesn't contains the preset values from constructor.
And the rootReducer will clone the action state in "RESET_TO".

My guess is that _initialState should contains the data of _preset* at line 297 (before the bind calls)

**Comment by masiorama** [comment]
I am experiencing the same issue. Did anybody find a solution?

**Comment by masiorama** [comment]
No feedback on this? There are other issues about this problem, all with no answers #1053

**Comment by masiorama** [comment]
> 

@yanmorinokamca I followed you analysis looking at the code and I guess you got it right... did you find any solution?

**Comment by masiorama** [comment]
As a temporary workaround in my application I managed to intercept reset event on the form and then repopulate all the options by hand:
```
document.getElementById('formId').addEventListener('reset', (ev) => {
    choices.setChoices(async () => {
        return this.options.map(({ value, label }) => ({
            value,
            label,
            selected: false,
        }));
    });
})
```
This should be unnecessary but at least my app does work. I hope this will be fixed at some point.

**Comment by yanmorinokamca** [comment]
> > 
> 
> @yanmorinokamca I followed you analysis looking at the code and I guess you got it right... did you find any solution?

Hi, we didn't have time to debug the library, so we replaced the form.reset() with a custom reset of each field.

**Comment by masiorama** [comment]
> > > 
> > 
> > 
> > @yanmorinokamca I followed you analysis looking at the code and I guess you got it right... did you find any solution?
> 
> Hi, we didn't have time to debug the library, so we replaced the form.reset() with a custom reset of each field.

I see. Unfortunately I cannot explicitly reset each field because it is a component that has to handle different forms, with many different fields.  Thanks for your reply.

**Comment by Sogl** [comment]
@masiorama I tried your code and saw that the event is not triggered:
```js

const choicesTagsObj = new Choices('#choices-tags', {
            maxItemCount: 4,
            removeItemButton: true,
            allowHTML: false,
            position: 'bottom',
            itemSelectText: 'Push',
            maxItemText: (maxItemCount) => {
                return `Only ${maxItemCount} can be selected`;
            },
        });

document.getElementById('choices-tags').addEventListener('reset', (ev) => {
     console.log('reset');
})
```



what's the problem?

**Comment by tagliala** [comment]
> this feature works fine in version 9.0.1, the bug has appeared since version 9.1.0

Bisected to: 68313da4122ec7f76274b22926e762914c96723d

(which is the conversion to TypeScript 🥲)

**Comment by Mopster** [comment]
I've encountered this issue now and hoping to see this fixed soon.
I tried the above temporary solution, but I lose my option groups.

**Comment by Xon** [comment]
Implemented as part of #1166
